scalardb-test/src/main/java/kelpie/scalardb/transfer/TransferProcessor.java

```diff
@@ -61,23 +80,49 @@ public void close() {
           builder.add(txId, Json.createArrayBuilder().add(ids.get(0)).add(ids.get(1)).build());
         });
 
-    setState(Json.createObjectBuilder().add("unknown_transaction", builder.build()).build());
+    setState(
+        Json.createObjectBuilder()
+            .add("num_updates", numUpdates.get())
+            .add("unknown_transaction", builder.build())
+            .build());
   }
 
-  private void transfer(DistributedTransaction transaction, int fromId, int toId, int amount)
+  private void transfer(
+      DistributedTransaction transaction,
+      int fromId,
+      int fromType,
+      int toId,
+      int toType,
+      int amount)
       throws TransactionException {
-    int fromType = 0;
-    int toType = 0;
-    if (fromId == toId) {
-      toType = 1; // transfer between the same account
+    int fromBalance = 0;
+    if (isWriteSkewTest.get()) {
+      int totalBalance = 0;
+      for (int i = 0; i < TransferCommon.NUM_TYPES; i++) {
+        Get get = TransferCommon.prepareGet(fromId, i);
+        Optional<Result> result = transaction.get(get);
+        int balance = TransferCommon.getBalanceFromResult(result.get());
+        if (i == fromType) {
+          fromBalance = balance;
+        }
+        totalBalance += balance;
+      }
+
+      if (totalBalance < 0) {
```

Comment:
Is this check only done in the processor ?
I think it would be better if it is done in the checker as well since inconsistent records won't be seen after updated.

Reply:
Thank you! The checker will check it again.